chain/icon/sender.go

```diff
@@ -145,6 +146,7 @@ func (s *sender) iconSegment(rm *chain.RelayMessage, height int64) ([]*chain.Seg
 		}
 		if len(msg.BlockUpdates) == 0 && len(msg.BlockProof) == 0 {
 			size += len(lbu.Proof)
+			// FIXME: this BlockUpdate.Proof is different from RelayMessage.BlockProof
```

Comment:
It should be correct if you do not segment the block updates by a limitation. Don't change this logic.

Reply:
The clause `len(msg.BlockUpdates) == 0` in Icon.Sender occurs only when BMR is restarted and sends a missing event.
Assume BMR has synced blocks from 5 to 10. However, it is still missing an event at block 8.

This scenario should not occur if a BMC has only one Relay.